Services/BookingService.cs

```diff
@@ -0,0 +1,82 @@
+﻿namespace BookStore.Services
+{
+    using BookStore.Data;
+    using BookStore.Interfaces;
+    using BookStore.Properties.Models;
+    using System.Collections.Generic;
+
+    public class BookingService : IBookingService
+    {
+        private readonly BookStoreContext _dbContext;
+
+        public BookingService(BookStoreContext context)
+        {
+            _dbContext = context;
+        }
+        public bool Delete(int id)
+        {
+            var booking = _dbContext.Booking.Find(id);
+            _dbContext.Booking.Remove(booking);
+
+            var user = _dbContext.User.Find(booking.User_Id);
+            user.Bookings.Remove(booking);
+
+            var status = _dbContext.Status.Find(booking.Status_Id);
+            status.Bookings.Remove(booking);
+
+            var product = _dbContext.Product.Find(booking.Product_Id);
+            product.Bookings.Remove(booking);
+
```

Comment:
You can get rid of this subsequent delete logic by implementing cascade deleting https://learn.microsoft.com/en-us/ef/core/saving/cascade-delete